packages/core/react/src/WalletProvider.tsx

```diff
@@ -157,13 +159,27 @@ export function WalletProvider({
             window.removeEventListener('beforeunload', handleBeforeUnload);
         };
     }, [adaptersWithStandardAdapters, walletName]);
+    const handleError = useCallback(
+        (walletError: WalletError) => {
+            (onError || console.error)(walletError);
+            if (
+                adapter &&
+                adapter.name !== SolanaMobileWalletAdapterWalletName &&
+                walletError.name === 'WalletConnectionError'
```

Comment:
Hmm this is too strict, a number of error types can be thrown by `connect` (account error, pubkey error, etc.) and any of them should trigger this.

The way we have the code in master, it's thrown whenever an error is encountered while connecting:
https://github.com/solana-labs/wallet-adapter/blob/afdc0b432150511d560a32aec1dcdcbdb8f7fbd3/packages/core/react/src/WalletProvider.tsx#L175-L181

Tricky here, because we want to special case MWA, and don't have access to `connecting` ref.

Reply:
Yeah, exactly. Really awkward to _directly_ wrap `connect()` the way the responsibilities are split here.

Only option I can think of is to invent a new `onConnectError` prop for `WalletProviderBase` to catch all manner of connection errors at the callsite.

Reply:
I guess I could also plumb through the `isConnecting` ref, instead. Hmm. Which is worse…